cmd/frontend/graphqlbackend/user_emails.go

```diff
@@ -12,18 +17,114 @@ import (
 	"github.com/sourcegraph/sourcegraph/cmd/frontend/envvar"
 	"github.com/sourcegraph/sourcegraph/internal/auth"
 	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/conf/deploy"
 	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/httpcli"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
 var timeNow = time.Now
 
-func (r *UserResolver) HasVerifiedEmail(ctx context.Context) (bool, error) {
+type HasVerifiedEmailResolver interface {
+	HasVerifiedEmail(ctx context.Context) (bool, error)
+}
+
+// dotcomUserHasVerifiedEmailResolver is a resolver for the hasVerifiedEmail field that should *only* be used by App.
+// Because use of Cody requires a dotcom account with a verified email, in app this sends a request
+// to sourcegraph.com to verify that the user associated with app has verified an email.
+type dotcomUserHasVerifiedEmailResolver struct {
+}
+
+// HasVerifiedEmail - checks with sourcegraph.com to ensure user has verified email.
+func (r *dotcomUserHasVerifiedEmailResolver) HasVerifiedEmail(ctx context.Context) (bool, error) {
+	// 🚨 SECURITY: This resolves HasVerifiedEmail only for App by
+	// sending the request to dotcom to check if a verified email exists for the user.
+	// Dotcom will ensure that only the authenticated user and site admins can check
+	if !deploy.IsApp() {
+		return false, errors.New("resolver only available in sourcegraph app")
+	}
+
+	if envvar.SourcegraphDotComMode() {
+		return false, errors.New("resolver not available")
+	}
+
+	// If app isn't configured with dotcom auth return false immediately
+	appConfig := conf.Get().App
+	if appConfig == nil {
+		return false, nil
+	}
+	if len(appConfig.DotcomAuthToken) <= 0 {
+		return false, nil
+	}
+
+	// If we have an app user with a dotcom authtoken ask dotcom if the user has a verified email
+	url := "https://sourcegraph.com/.api/graphql?AppHasVerifiedEmailCheck"
+	cli := httpcli.ExternalDoer
+	payload := strings.NewReader("{\"query\":\"query AppHasVerifiedEmailCheck{ currentUser { hasVerifiedEmail } }\",\"variables\":{}}")
```

Comment:
Maybe use a raw string literal?

```suggestion
	payload := strings.NewReader(`{"query":"query AppHasVerifiedEmailCheck{ currentUser { hasVerifiedEmail } }","variables":{}}`)
```